lib/auth/auth_with_roles.go

```diff
@@ -2328,18 +2329,54 @@ func (a *ServerWithRoles) CreateSAMLAuthRequest(req services.SAMLAuthRequest) (*
 		return nil, trace.Wrap(err)
 	}
 
+	// require additional permissions for executing SSO test flow.
+	if req.SSOTestFlow {
+		if err := a.authConnectorAction(apidefaults.Namespace, types.KindSAML, types.VerbCreate); err != nil {
```

Comment:
Why does SSO test require "create" permissions for the connector? If you're running a test, you're not actually creating the connector, right?

Reply:
Security perspective: we want the permissions to be high enough to argue the user is not gaining too much power by running the test.